x-pack/plugins/security_solution/public/detections/components/alerts_kpis/alerts_histogram_panel/index.tsx

```diff
@@ -174,20 +181,30 @@ export const AlertsHistogramPanel = memo<AlertsHistogramPanelProps>(
     }, [defaultStackByOption, onlyField]);
 
     const { toggleStatus, setToggleStatus } = useQueryToggle(DETECTIONS_HISTOGRAM_ID);
-    const [querySkip, setQuerySkip] = useState(!toggleStatus);
+    const [querySkip, setQuerySkip] = useState(
+      isAlertsPageChartsEnabled ? !isExpanded : !toggleStatus
+    );
     useEffect(() => {
-      setQuerySkip(!toggleStatus);
-    }, [toggleStatus]);
+      if (isAlertsPageChartsEnabled && isExpanded !== undefined) {
```

Comment:
This code seems to be exactly similar to `x-pack/plugins/security_solution/public/detections/components/alerts_kpis/alerts_count_panel/index.tsx`.

Do you think it should be extracted to separate hook.